handlers/main.yml

```diff
@@ -27,20 +27,12 @@
   command: /bin/true
   notify:
     - Stop docker
-    - Destroy docker0
     - Start docker
 
 - name: Stop docker
   become: yes
   service: name=docker.service state=stopped
 
-- name: Destroy docker0
```

Comment:
We can not completely remove this task - this is required for new setups when docker and flannel are installed for the first step. It might be OK not to delete the docker bridge only in the case of flannel upgrades when we don't expect flannel to change the node's subnet.

Reply:
I think we're okay here. This handler was removing the existing docker0 bridge to clear out all the existing containers associated with it. Before docker and flannel are installed for the first time there would be no containers running or associated with docker0 bridge. Once we install and reconfigure docker and flannel with ansible, all the new containers will be correctly associated with docker0 bridge by default. I've tested this by creating a brand new cluster and then running ansible with this change.

Reply:
If flannel is restarted there's a possibility that it can get a new subnet (not always) for the node, different from the one the node already had. In that case it's mandatory that the docker bridge is recreated and docker is restarted without live-restore option so that all the containers in the node are on the new subnet. If that's not done, we'll end up with old non-reachable containers on the node. So recreating docker bridge should be the default behavior.

But we could make that task conditional, if there's no change in the subnet after a flannel restart or upgrade.

Reply:
We should collapse `Reconfigure docker`, `Stop docker` and `Start docker` to just one handler called `Restart docker`.

Reply:
@naveens Like discusses offline, I am keeping this change and removing the docker0 bridge clean-up code. Flannel will always use the same subnet unless we change the node's public_ip and remove the subnet record in ectd and remove the `/var/run/flannel/subnet.env` file. We've the handler to remove remove `/var/run/flannel/subnet.env` file conditionally, I am keeping that since the condition is set the false by default. To summarize we can be sure that the Flannel will always pick up same subnet and we should not be deleting the docker bridge in any case.

Reply:
@alex-vmw Optimized the docker handler, please review.